File: Web/src/app/models/player-opponent.ts
import { Timestamp } from 'firebase/firestore';

export interface PlayerOpponents {
    Id:string;
    FullName:string;
    Opponents:PlayerOpponent[];
}

export interface PlayerOpponent {
    Id:string;
    FullName:string;
    Matches:PlayerOpponentMatch[];
}

export interface PlayerOpponentMatch{
    Id:string;
    MatchDate:Timestamp;
    WinnerId:string;
    LoserId:string;
    WinnerSetWins:number;
    WinnerSetScores:number[];
    LoserSetWins:number;
    LoserSetScores:number[];
    
}

File: Importers/TtPlayers.Importer/Domain/Models/PlayerOpponent.cs
using Google.Cloud.Firestore;

namespace TtPlayers.Importer.Domain.Models
{
    [FirestoreData]
    public class PlayerOpponent
    {
        [FirestoreProperty]
        public string? Id { get; set; }

        [FirestoreProperty]
        public string? FullName { get; set; }

        [FirestoreProperty]
        public List<PlayerOpponentMatch> Matches { get; set; }
    }

    [FirestoreData]
    public class PlayerOpponentMatch
    {
        [FirestoreProperty]
        public string? Id { get; set; }

        [FirestoreProperty]
        public DateTime MatchDate { get; set; }

        [FirestoreProperty]
        public string WinnerId { get; set; }

        [FirestoreProperty]
        public string LoserId { get; set; }

        [FirestoreProperty]
        public int? WinnerSetWins { get; set; }

        [FirestoreProperty]
        public List<int> WinnerSetScores { get; set; }

        [FirestoreProperty]
        public int? LoserSetWins { get; set; }

        [FirestoreProperty]
        public List<int> LoserSetScores { get; set; }

    }
}
